src/HeadersBuilder.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Minishlink\WebPush;
+
+use Base64Url\Base64Url;
+use ErrorException;
+
+class HeadersBuilder
+{
+    /**
+     * @param Notification $notification
+     *
+     * @return Headers
+     * @throws ErrorException
+     */
+    public function build(Notification $notification): Headers
+    {
+        $headers = new Headers();
```

Comment:
This method is quite complex. Refactoring should be done here.

Reply:
I agree, the entire class actually. This was the easiest on the eyes I could make it without getting too deep into it. Nonetheless, it is abstracted away behind this method and can be cleaned up freely 🙂